src/main/java/com/github/shyiko/mysql/binlog/event/deserialization/AbstractRowsEventDataDeserializer.java

```diff
@@ -524,17 +898,27 @@ protected int deserializeFractionalSeconds(int meta, ByteArrayInputStream inputS
         return 0;
     }
 
+    protected int deserializeFractionalSeconds(int meta, BinaryLogEventDataReader eventDataReader) {
+        int length = (meta + 1) / 2;
+        if (length > 0) {
+            int fraction = eventDataReader.readIntegerBE(length);
+            return fraction * (int) Math.pow(100, 3 - length);
+        }
+        return 0;
+    }
+
     private static int bitSlice(long value, int bitOffset, int numberOfBits, int payloadSize) {
         long result = value >> payloadSize - (bitOffset + numberOfBits);
         return (int) (result & ((1 << numberOfBits) - 1));
     }
 
     private static int numberOfBitsSet(BitSet bitSet) {
-        int result = 0;
-        for (int i = bitSet.nextSetBit(0); i >= 0; i = bitSet.nextSetBit(i + 1)) {
-            result++;
-        }
-        return result;
+        return bitSet.cardinality();
+//        int result = 0;
+//        for (int i = bitSet.nextSetBit(0); i >= 0; i = bitSet.nextSetBit(i + 1)) {
+//            result++;
+//        }
+//        return result;
```

Comment:
is it intentional to retain this commented code? can you explain the difference between the bitset cardinality function vs this prior implementation?

Reply:
No, that's a dead code I forgot to remove.

Reply:
Removed